### PR TITLE
Convert assert to exception for loading native libraries

### LINKED ISSUE #70293: Assert failure: !LongFile::IsPathNotFullyQualified(path) || !LongFile::ContainsDirectorySeparator(path)
PMI of Interop\DllImportAttribute\DllImportPath\DllImportPathTest\DllImportPathTest.dll causes:

```
Assert failure(PID 78956 [0x0001346c], Thread: 46696 [0xb668]): !LongFile::IsPathNotFullyQualified(path) || !LongFile::ContainsDirectorySeparator(path)
CORECLR! LoadLibraryExWrapper + 0x1A0 (0x7b1e01fd)
CORECLR! CLRLoadLibraryEx + 0x29 (0x7ae80ecf)
CORECLR! `anonymous namespace'::LocalLoadLibraryHelper + 0xF1 (0x7afdc26e)
CORECLR! `anonymous namespace'::LoadNativeLibraryBySearch + 0x2D7 (0x7afdb2d7)
CORECLR! `anonymous namespace'::LoadNativeLibraryBySearch + 0xB0 (0x7afdb5a0)
CORECLR! `anonymous namespace'::LoadNativeLibrary + 0x453 (0x7afdaeb4)
CORECLR! NativeLibrary::LoadLibraryFromMethodDesc + 0x303 (0x7afda6d5)
CORECLR! `anonymous namespace'::NDirectLink + 0x2C0 (0x7aeff96c)
CORECLR! NDirect::GetStubForILStub + 0x225 (0x7aefb9cf)
CORECLR! GetStubForInteropMethod + 0x27D (0x7aefbc65)
    File: C:\gh\runtime\src\coreclr\utilcode\longfilepathwrappers.cpp Line: 64    Image: c:\gh\runtime\artifacts\tests\coreclr\windows.x86.Checked\Tests\Core_Root\corerun.exe
```

e.g.,
```
pushd c:\gh\runtime\artifacts\tests\coreclr\windows.x86.Checked\Tests\Core_Root
corerun.exe c:\gh\jitutils\bin\pmi.dll DRIVEALL c:\gh\runtime\artifacts\tests\coreclr\windows.x86.Checked\Interop\DllImportAttribute\DllImportPath\DllImportPathTest\DllImportPathTest.dll > pmi-DllImportPathTest-x86.log
popd
```

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Fixes #70293

/cc @dotnet/interop-contrib

### CODE REVIEW

File: src/coreclr/utilcode/longfilepathwrappers.cpp [hunk 1]
```diff
@@ -54,14 +54,15 @@ LoadLibraryExWrapper(
...
     {
-
         LongPathString path(LongPathString::Literal, lpLibFileName);
 
         if (LongFile::IsPathNotFullyQualified(path) || SUCCEEDED(LongFile::NormalizePath(path)))
         {
 #ifdef HOST_WINDOWS
-            //Adding the assert to ensure relative paths which are not just filenames are not used for LoadLibrary Calls
-            _ASSERTE(!LongFile::IsPathNotFullyQualified(path) || !LongFile::ContainsDirectorySeparator(path));
+            // Ensure relative paths which are not just filenames are not used for LoadLibrary calls.
```
Comment (jkotas) [inline src/coreclr/utilcode/longfilepathwrappers.cpp:62]: Is this a breaking change? Are there cases that worked just fine in release builds before this change and that are going to fail with exception now?

Comment (AaronRobinsonMSFT) [reply src/coreclr/utilcode/longfilepathwrappers.cpp:62]: I don't think so, but that is a concern. I believe the tricky bit here was having relative path that couldn't be resolved to a real path anyways. The inner message might be different but the public facing exception type should remain the same.

Comment (jkotas) [reply src/coreclr/utilcode/longfilepathwrappers.cpp:62]: I think this is breaking change. I have done some ad-hoc testing and the relative paths worked just fine before this change as far as I can tell.

For example, try `NativeLibrary.Load(@"..\windows.x64.Release\clrjit_win_x86_x64.dll")`, with `clrjit_win_x86_x64.dll` in appropriate place.

Delete the assert and regression test instead?

File: src/tests/Interop/NativeLibrary/API/NativeLibraryTests.cs [hunk 2]
```diff
@@ -46,6 +46,23 @@ public void LoadLibraryOnInvalidFile_NameOnly()
         EXPECT(TryLoadLibrary_NameOnly(libName), TestResult.ReturnFailure);
     }
 
+    [Fact]
+    [PlatformSpecific(TestPlatforms.Windows)]
```
Comment (elinor-fung) [inline src/tests/Interop/NativeLibrary/API/NativeLibraryTests.cs:50]: Why Windows-only? Shouldn't non-Windows also throw on an invalid relative path and work with a valid one?

**Comment by AaronRobinsonMSFT** [comment]
@elinor-fung or @jkotas It is green now. Any other comments or concerns?

**Review by jkotas — APPROVED** [review]

**Pull request merged**